### ISSUE TITLE
relocateClass should not be called with resource paths

### ISSUE THREAD

**Issue by oehme** [GradleUp/shadow#867]
The following line should call `relocateClass` with just the file name, like in the if-statement above and take care of the `META-INF/services` prefix itself. The `relocateClass` method shouldn't have to deal with slashes. 

https://github.com/johnrengelman/shadow/blob/9c5182d2d9c5f7141e4d2a525ec94d6111283cd9/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/ServiceFileTransformer.groovy#L75 

Noticed this while implementing my own relocator.

**Comment by Goooler** [comment]
It keeps the same logic from https://github.com/apache/maven-shade-plugin/blob/89bdd78ca2f4756a83eee5c3d2b437011c0b8d7f/src/main/java/org/apache/maven/plugins/shade/resource/ServicesResourceTransformer.java#L61.

**Comment by oehme** [comment]
No, look just a [few lines up](https://github.com/apache/maven-shade-plugin/blob/89bdd78ca2f4756a83eee5c3d2b437011c0b8d7f/src/main/java/org/apache/maven/plugins/shade/resource/ServicesResourceTransformer.java#L58) in the code you linked. It removes the META-INF prefix.

**Comment by Goooler** [comment]
The latest code matches what you want?

https://github.com/GradleUp/shadow/blob/05a47b55ba420faea46e3d1ff61c9d39a3b58700/src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/ServiceFileTransformer.kt#L48-L49

**Comment by oehme** [comment]
Yes, indeed. Thanks!
